lib/pages/settings/view/admin_page.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/settings/service/admin_provider.dart';
+import 'package:acs_upb_mobile/pages/settings/view/request_card.dart';
+import 'package:acs_upb_mobile/widgets/scaffold.dart';
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_feather_icons/flutter_feather_icons.dart';
+import 'package:provider/provider.dart';
+
+class AdminPanelPage extends StatefulWidget {
+  const AdminPanelPage({Key key}) : super(key: key);
+  static const String routeName = '/admin';
+
+  @override
+  _AdminPanelPageState createState() => _AdminPanelPageState();
+}
+
+class _AdminPanelPageState extends State<AdminPanelPage> {
+  Future<List<String>> requests;
+  List<String> requestIds;
+  bool all = false;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final adminProvider = Provider.of<AdminProvider>(context);
+    if (!all) {
+      requests = adminProvider.fetchUnprocessedRequests();
+    } else {
+      requests = adminProvider.fetchAllRequests();
+    }
+    return AppScaffold(
+      actions: [
+        AppScaffoldAction(
+            icon: FeatherIcons.filter,
+            tooltip: S.current.navigationFilter,
+            items: {
+              S.current.filterMenuShowUnprocessed: () {
+                if (all) {
+                  requests = adminProvider.fetchUnprocessedRequests();
```

Comment:
This is unnecessary, you're already making this check and fetch at the beginning of the `build` method.